apps/gitness/src/pages-v2/project/project-rules-create-container.tsx

```diff
@@ -0,0 +1,208 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useNavigate, useParams } from 'react-router-dom'
+
+import {
+  useListPrincipalsQuery,
+  useListStatusCheckRecentSpaceQuery,
+  useSpaceRuleAddMutation,
+  useSpaceRuleGetQuery,
+  useSpaceRuleUpdateMutation
+} from '@harnessio/code-service-client'
+import { PrincipalType } from '@harnessio/ui/types'
+import {
+  BranchRulesActionType,
+  getBranchRules,
+  MergeStrategy,
+  RepoBranchSettingsFormFields,
+  RepoBranchSettingsRulesPage
+} from '@harnessio/ui/views'
+
+import { useGetSpaceURLParam } from '../../framework/hooks/useGetSpaceParam'
+import { useMFEContext } from '../../framework/hooks/useMFEContext'
+import { useTranslationStore } from '../../i18n/stores/i18n-store'
+import { transformDataFromApi, transformFormOutput } from '../../utils/repo-branch-rules-utils'
+import { useBranchRulesStore } from '../repo/stores/repo-branch-rules-store'
+import { useProjectRulesStore } from './stores/project-rules-store'
```

Comment:
i see we added a `projectRulesStore`. why do we need to use `branchRulesStore` here from repoRules?

Reply:
wait, I think I'm mixing a few things. maybe the naming is not ideal.
I see we have `repoRules`, `branchRules` and `projectRules`. How do they relate to each other?